cpp/src/arrow/util/mutex.cc

```diff
@@ -35,9 +36,12 @@ struct Mutex::Impl {
 
 Mutex::Guard::Guard(Mutex* locked)
     : locked_(locked, [](Mutex* locked) {
+#if !defined(__EMSCRIPTEN__) || defined(ARROW_ENABLE_THREADING)
```

Comment:
Is this `__EMSCRIPTEN__` check redundant?

```suggestion
#ifdef ARROW_ENABLE_THREADING
```